### PR TITLE
test: phase4

### PR DESCRIPTION
**Pull request by nugaon**
Tailoring test cases to phase4 changes and mine samples for stats tests and added phase4 related checks

### CODE REVIEW

**Review by zelig — COMMENTED** [review]
please make stats fixtures less files and/or number them 001,002 etc
and submit as a separate PR so that we can see logic changes better

File: src/Util/ChunkProof.sol [hunk 1]
```diff
@@ -43,7 +43,7 @@ library BMTChunk {
   ) internal pure returns (bytes32 _calculatedHash) {
     _calculatedHash = _proveSegment;
     for (uint256 i = 0; i < _proofSegments.length; i++) {
-      bool mergeFromRight = _proveSegmentIndex % 2 == 0 ? true : false;
```
[the remark this answers is not in the document]
Comment (nugaon) [reply src/Util/ChunkProof.sol:46]: this was just only a bit of refactor in the code, but it does the same thing.
if it should be delivered in a different PR, I can revert that.

File: test/util/tools.ts [hunk 2]
```diff
@@ -180,7 +159,7 @@ export async function mineToRevealPhase() {
  */
 export async function copyBatchForClaim(
   deployer: string
-): Promise<{ tx: any; postageDepth: number; initialBalance: number }> {
```
[the remark this answers is not in the document]
Comment (nugaon) [reply test/util/tools.ts:183]: refactored

**Comment by nugaon** [comment]
> please make stats fixtures less files and/or number them 001,002 etc
> and submit as a separate PR so that we can see logic changes better

If I do that, the CI starts to mine again the samples that are missing.

All samples are located in a separated folder under the `test/mined-witnesses` , so I kindly ask you to ignore the changes in that folder. 
The different round samples are numbered and prefixed with the related test name e.g. `stats-0.json`, `stats-1.json` and so on.

**Pull request merged**